### ISSUE TITLE
Move dockerfiles of open source components to another repos

### ISSUE THREAD

**Issue by MartinForReal** [goharbor/harbor#9213]
**Is your feature request related to a problem? Please describe.**
Update base image and opensource components to mitigate security risks in dependencies and base images. 

**Describe the solution you'd like**
move make/photon/{chartserver,clair,notary,redis,registry,db} to another repository and use CI/CD saas service (such as travis-ci,circle ci) to build these images daily in order to get the most updated version of these components. 

**Describe the main design/architecture of your solution**
move make/photon/{chartserver,clair,notary,redis,registry,db} to another repository

**Describe the development plan you've considered**
 db container to use docker volume and adopt db schema migration tool
all of container to mount generated ca certificate into /etc/ssl/certificates (following the instructions [here](https://medium.com/@paraspatidar/add-self-signed-or-ca-root-certificate-in-kubernetes-pod-ca-root-certificate-store-cb7863cb3f87))

**Additional context**
related issue: 
https://github.com/goharbor/harbor/issues/6040
https://github.com/goharbor/harbor/issues/8790

**Comment by reasonerjt** [comment]
@MartinForReal 
As long as we package these images in installer, I don't see why moving it to another repo solves any problem, could you elaborate the problem you are seeing with current approach?

**Comment by MartinForReal** [comment]
We have source code and dependency config in the single repository and we packed some customization into containers (such as create DB schema, replace certificate bundle file). 
which makes it hard to leverage automation tools to eliminate security vulnerabilities in dependencies(base image, middleware, and others).  By moving to another repo, we can split the customization configuration from base images. In this way, we can update base images with less effort and end-users who use external Redis, pg and other middleware would get more tools and documents.  if we found any security issue, we don't need to wait until official images are released. [ref](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get)

Also, we don't have a clearly defined release process and don't know the exact version of each dependency which makes it hard to investigate and mitigate security issues. For example, I have to download all images in order to generate a list of component versions and this makes it harder to draw people's attention when security vulnerabilities are found.
